cmd/restic/cmd_cache.go

```diff
@@ -130,3 +148,14 @@ func runCache(opts CacheOptions, gopts GlobalOptions, args []string) error {
 
 	return nil
 }
+
+func dirSize(path string) (int64, error) {
+	var size int64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
```

Comment:
You need to check `err` first, return it if it's non-nil, then check if `info` is nil (which may happen) and return otherwise. and only afterwards access `info.Size()`.